### PR TITLE
Remove the pre logic for empty invokers.

### PR DESCRIPTION
**Pull request by carryxyh**
Enhance for issue https://github.com/apache/incubator-dubbo/issues/1861

Remove the pre logic for empty invokers.

### CODE REVIEW

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/loadbalance/AbstractLoadBalance.java [hunk 1]
```diff
@@ -45,12 +44,6 @@ static int calculateWarmupWeight(int uptime, int warmup, int weight) {
 
     @Override
     public <T> Invoker<T> select(List<Invoker<T>> invokers, URL url, Invocation invocation) {
-        if (CollectionUtils.isEmpty(invokers)) {
-            return null;
-        }
-        if (invokers.size() == 1) {
-            return invokers.get(0);
-        }
```
Comment (scxwhite) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/loadbalance/AbstractLoadBalance.java]: I have a puzzle about why you put these empty check codes into subclasses, Can you explain that?

Comment (carryxyh) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/loadbalance/AbstractLoadBalance.java]: To make all the `select` logic extendable.

**Comment by zonghaishang** [comment]
I left my opinion, please take a moment to look at that https://github.com/apache/incubator-dubbo/issues/1861

**Comment by carryxyh** [comment]
Hi, yiji, nice work.
I'd like close this pr. Let's focus on the issue comments.

**Pull request closed without merging**